src/client/api/container.c

```diff
@@ -156,6 +156,30 @@ daos_cont_query(daos_handle_t coh, daos_cont_info_t *info,
 	return dc_task_schedule(task, true);
 }
 
+int
+daos_cont_get_acl(daos_handle_t coh, daos_prop_t **acl_prop, daos_event_t *ev)
+{
+	daos_prop_t	*prop;
+	const size_t	nr_entries = 3;
+
+	if (acl_prop == NULL) {
+		D_ERROR("invalid acl_prop parameter\n");
+		return -DER_INVAL;
+	}
+
+	prop = daos_prop_alloc(nr_entries);
+	if (prop == NULL)
+		return -DER_NOMEM;
+
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_ACL;
+	prop->dpp_entries[1].dpe_type = DAOS_PROP_CO_OWNER;
+	prop->dpp_entries[2].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+
+	*acl_prop = prop;
+
+	return daos_cont_query(coh, NULL, prop, ev);
```

Comment:
shouldn't you free prop if query fails?

Reply:
Good catch, will fix.

Reply:
Well, space will be freed soon upon exit due to error. I still ask myself each time if this cleanup code really needs to be present... Only for the client/command case for sure ;-)  At least, it is required to make Valgrind happy ??